### ISSUE TITLE
could you  provide  a user guide to use the code in AS ? 

### ISSUE THREAD

**Issue by gogov5** [FangGet/ORB_SLAM2_Android#3]
it is a great work, could you provide a user guide to use the code in AS?  thanks

**Comment by gogov5** [comment]
Hi  Frank, I run this code on Android phone, only can achieve  1.3 FPS, and  often fail to init (use kitti dataset).  do you have the same performance in your test ?  thank you

**Comment by aivijay** [comment]
gogov, can you share the .yaml file you used to test or how you calibrated your phone camera. I am trying to just see how this works in the phone compared to LSD SLAM.

**Comment by gogov5** [comment]
hi aivijay，
you can use your phone camera to get the pic of chessboard, then use opencv or matlab tool to do calibration.  and get fx fy cx cy in the ymal file.   note that the resolution setting.

发送自我的三星 Galaxy 智能手机。
-------- 原始信息 --------由： aivijay notifications@github.com 日期: 16/10/17  09:19  (GMT+08:00) 收件人： FangGet/ORB_SLAM2_Android ORB_SLAM2_Android@noreply.github.com 抄送： gogov5 guobin6898@163.com, Author author@noreply.github.com 主题： Re: [FangGet/ORB_SLAM2_Android] could you provide a user guide to use
  the code in AS ? (#3) 
gogov, can you share the .yaml file you used to test or how you calibrated your phone camera. I am trying to just see how this works in the phone compared to LSD SLAM.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/FangGet/ORB_SLAM2_Android","title":"FangGet/ORB_SLAM2_Android","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/FangGet/ORB_SLAM2_Android"}},"updates":{"snippets":[{"icon":"PERSON","message":"@aivijay in #3: gogov, can you share the .yaml file you used to test or how you calibrated your phone camera. I am trying to just see how this works in the phone compared to LSD SLAM.\r\n\r\n"}],"action":{"name":"View Issue","url":"https://github.com/FangGet/ORB_SLAM2_Android/issues/3#issuecomment-254091839"}}}

**Comment by aivijay** [comment]
I got the app working on the phone, though its quite slow with the frame rate. I had similar problems with LSD SLAM on a laptop itself. I will try to see if I can get a better calibration done sometimes soon and see how that goes. Thanks for your help. I could never get the dataset based slam working as it was quitting after quite sometimes. I tried with the TUM dataset though.

**Comment by aivijay** [comment]
I managed to build the android build from my machine and got it installed and tried. Maybe I can play around with somethings now as I can do my own builds.

**Comment by tiejiang** [comment]
I have cloned the project into my AS IDE, and use NDK to configuration, but I do not know why it was not worked. Have you ever met this problem?

**Comment by pisuto** [comment]
So anyone can share the method of how to use AS for this program?

**Comment by tomsfernandez** [comment]
Hi, has anyone been able to succesfully use AS. If they have, I would appreciate if they could post it

**Comment by bouviervj** [comment]
Hi, I'm using AS currently with that project. I did a conversion from Eclipse to AS ... and I'm currently building configurations so that it could compile correctly. Unfortunately for now I was not able to run Frank version or mine - it crashes when loading XML description of the Android app - I will let you know if I'm able to share an AS version.

**Comment by IanBoyan** [comment]
@bouviervj I am also trying to get the project working on AS. So far I am able to load the UI under camera mode. However, OpenCV loading always fails. Have you encountered similar issue? Thanks!

**Comment by BrianT34** [comment]
@bouviervj I am also trying to get the project working on AS. So far I am able to load the UI under camera mode. However, OpenCV loading always fails. Have you encountered similar issue? Thanks!

**Comment by bouviervj** [comment]
I had that issue, I dealt with it by migrating to a new version of opencv that matches the opencv manager I had on my android phone.

**Comment by BrianT34** [comment]
Thanks! I am trying to use newer 331 version of it. No luck yet, will come back later to share how it goes. Are you able to finally resolve all the issues?

**Comment by bouviervj** [comment]
Yup I got the algorithm to work in camera mode. AS was really slow to compile on windows platform.
![screenshot_20180102-131042](https://user-images.githubusercontent.com/3627391/34494567-2900751e-efbf-11e7-9dbb-96f3c8a05cc6.jpg)

**Comment by BrianT34** [comment]
This is awesome! Just get chance to finish importing OpenCV331 and/or OpenCV2411. Still no luck. Same issue when using async loading. The app crashes. Would you mind to share your gradle file? Thanks!

**Comment by bouviervj** [comment]
@BrianTan34 I'll try to create a branch on my repo as as soon as possible, so that we could work on that AS migration. I planned to do it but unfortunately modifications were not clean (hard path etc...)

**Comment by Zander-yang** [comment]
@bouviervj  If you feel it slowly to compile in AS, you can use ndk-build to build so first. Then import *.so in AS.
I tried operating dataset mode with TUM dataset. It can't generate map when i use the TUM1.yaml. It could when I change the fps from 30 to 5. 
Can you operate smoothly?  Which phone do you used?

**Comment by bouviervj** [comment]
My phone seems (seems) to operate smoothly, it's an Asus ZenPhone 3 Model # ASUS_Z01FD .

**Comment by guardfromua** [comment]
@bouviervj how did you manage to work it in AS. Can you share the algorithm?

**Comment by bouviervj** [comment]
Hi,

I finally pushed a repo on github for the AS version, just let me know if you're able to compile it.

[ORB_SLAM2_AndroidStudio](https://github.com/bouviervj/ORB_SLAM2_AndroidStudio)

Cheers,
Julien

**Comment by guardfromua** [comment]
Thank you a lot!
+ in karma!)
